index.js

```diff
@@ -0,0 +1,47 @@
+// Copyright IBM Corp. 2014,2016. All Rights Reserved.
+// Node module: loopback-soap
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+var SoapGenerator = require('./lib/codegen/generator-soap');
+
+
+function getGenerator() {
+  var generator = new SoapGenerator();
+  return generator;
+}
+
+/**
+ * Generate remote methods from soap spec
+ * @param {Object} spec
+ * @param {Object} options
+ * @returns {String}
+ */
+exports.generateRemoteMethods = function(spec, options) {
+  return getGenerator(spec).generateRemoteMethods(spec, options);
+};
+
+/**
+ * Generate remote methods for an array of operations
+ * @param {String|Number} version
+ * @param {String} modelName
+ * @param {BaseOperation[]} operations
+ * @returns {String}
+ */
+exports.generateCode = function(version, modelName, operations) {
+  return getGenerator(spec).generateCodeForOperations(modelName, operations);
```

Comment:
`spec` is not defined. Should the method signature be `exports.generateCode = function(spec, modelName, operations)`?